### ISSUE TITLE
Upload: interruptUpload throws AssertionError

### ISSUE THREAD

**Issue by willemv** [vaadin/framework#11682]
Vaadin Framework version: 8.8.5
Browser: any
Web container: any

Caused by the fix for #11616
Related issues: #4212 #9635

Sample project to reproduce the issue: https://github.com/willemv/vaadin-upload-bug

To reproduce:
* start a Vaadin-based application with assertions enabled (i.e. by adding the `-ea` JVM parameter)
* create an `Upload` button
* start uploading a file
* call the `interruptUpload` method on the upload button

Result: an `AssertionError` is thrown, because the `FileUploadHandler` tries to access the connector state outside of the session lock:

```
Jul 31, 2019 3:50:17 PM com.vaadin.server.DefaultErrorHandler doDefault
SEVERE: 
java.lang.AssertionError: Session must be locked when getState() is called
	at com.vaadin.server.AbstractClientConnector.getState(AbstractClientConnector.java:264)
	at com.vaadin.ui.AbstractComponent.getState(AbstractComponent.java:676)
	at com.vaadin.ui.AbstractSingleComponentContainer.getState(AbstractSingleComponentContainer.java:364)
	at com.vaadin.ui.UI.getState(UI.java:340)
	at com.vaadin.ui.PushConfigurationImpl.getState(PushConfiguration.java:336)
	at com.vaadin.ui.PushConfigurationImpl.getPushMode(PushConfiguration.java:181)
	at com.vaadin.server.communication.FileUploadHandler.streamToReceiver(FileUploadHandler.java:621)
	at com.vaadin.server.communication.FileUploadHandler.handleFileUploadValidationAndData(FileUploadHandler.java:460)
	at com.vaadin.server.communication.FileUploadHandler.doHandleSimpleMultipartFileUpload(FileUploadHandler.java:413)
	at com.vaadin.server.communication.FileUploadHandler.handleRequest(FileUploadHandler.java:290)
	at com.vaadin.server.VaadinService.handleRequest(VaadinService.java:1602)
	at com.vaadin.server.VaadinServlet.service(VaadinServlet.java:445)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:841)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:535)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:561)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:334)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:104)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:243)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:679)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:597)
	at java.lang.Thread.run(Thread.java:748)
```
If you run _without_ assertions, the cancel simply doesn't work

**Comment by TatuLund** [comment]
This issue could be related https://github.com/vaadin/framework/issues/10545

**Comment by willemv** [comment]
Indeed, it seems related to #10545. If the streamVariable is not cleaned, I get the same buggy behavior as in that bug, but with the fixes in PR #11743 that bug is fixed as well

**Comment by TatuLund** [comment]
@willemv Thanks for your effort on testing that.
